### ISSUE TITLE
Amazon Digital Music pages fail to extract

### ISSUE THREAD

**Issue by ROpdebee** [ROpdebee/mb-userscripts#611]
So Amazon overhauled their digital music pages on the main site. E.g. https://www.amazon.com/dp/B07R92TVWN. Everything is dynamically loaded through an API now (https://na.mesk.skill.music.a2z.com/api/showHome), with (seemingly) no way of grabbing cover art without also going through the API. The API appears to block any request that doesn't include like 20 custom headers, also requiring a bunch of unique data  (CSRF tokens, request IDs, session IDs, etc). At first glance it looks possible to create these headers, since it appears that unique data is available in the HTML page that we already load. However, the images that the API returns are limited to 500x500, and higher resolution images like the ones we were getting previously don't seem to be served anymore. Thus, I'm on the fence about whether the required effort to support these new pages would even be worth it, assuming that most (if not all) Amazon Music releases are available on other streaming services with higher resolution cover art.

Some alternative solutions:
* Use a seeding button like on a-tisket for those pages.
* Do some trickery with `window.open` and `window.postMessage` so that ECAU lets a new tab/popup load the image, runs on that tab, and communicates the image URL back to the MB page.

TL;DR: Amazon has made it very difficult to extract cover art from Amazon Music releases, and the artwork is of such low quality that I doubt that adding support for those pages is worth it. Maybe we should just fail gracefully (instead of "Failed to extract images from generic page") and tell the user to use a different source.

Since I don't use Amazon as a source a lot, I'm not sure whether it's generally true that these releases are available elsewhere or whether it's even acceptable to drop support for Amazon Music. @hibiscuskazeneko and @kellnerd, do you have any insights? Or do you possibly know of other means of getting these images?

**Comment by kellnerd** [comment]
> Since I don't use Amazon as a source a lot, I'm not sure whether it's generally true that these releases are available elsewhere or whether it's even acceptable to drop support for Amazon Music.

I never had to rely on Amazon for the artwork of digital releases so far, usually there are multiple other websites which have better artwork for them. So it would be totally acceptable for me to drop support for Amazon Music. Personally I only use Amazon for audiobook covers (Audible sometimes has the best images for them) or as a fallback for CD cover art when other websites don't provide it and I'm too lazy/unable to scan it.

**Comment by HibiscusKazeneko** [comment]
Same here. I only use Amazon as a source for cover art for physical releases.
I guess this means we're (partially?) SOL for Amazon Music-exclusive releases :/

**Comment by ROpdebee** [comment]
If Amazon Music exclusives become common place, let me know and I'll look into supporting them again or finding an acceptable workaround. For now, copying the image URL should work if there's truly no other provider.
